applications/Substrate_Move_System_Pallet_1.md

```diff
@@ -0,0 +1,185 @@
+
+## Development Status **📖**
+
+- This application is a direct response to [this RFP](https://github.com/w3f/Grants-Program/blob/master/docs/RFPs/Open/move_smart_contract_pallet.md)
+- [Move language Github](https://github.com/move-language/move/tree/main/language)
+- [Pontem MoveVM fork](https://github.com/pontem-network/sp-move-vm)
+- [Pontem MoveVM system pallet](https://github.com/pontem-network/pontem/tree/master/pallets/sp-mvm)
+
+We have not yet started our own development, we are still in the research phase.
+
+## Development Roadmap **🔩**
+
+### Overview
+
+- **Total Estimated Duration:** 1 month
+- **Full-Time Equivalent (FTE):** 2 FTE
+- **Total Costs:** 48 000 USD
+
+### **In-Depth Exploration and Assessment of MoveVM and Substrate Integration**
+
+**Goal**: Research Pontem Move VM solution, Move language and its ecosystem, and document all findings. Prepare a repository for developing the Substrate Move system pallet. 
+
+| Number | Deliverable | Specification |
+| --- | --- | --- |
+| 0a. | License | Apache 2.0 and MIT |
+| 0b. | Documentation | We will provide both inline documentation of the code and a basic tutorial that explains how a user can (for example) spin up one of our Substrate nodes and send test transactions, which will show how the new functionality works. |
+| 0c. | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| 0d. | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 0e. | Article | Content: article that explains all of the research and findings done in the research phase, and how it shapes the project in the future. The research would include: <ul><li>analysis of the Pontem Move fork,</li><li>analysis of the Pontem MoveVM system pallet, evaluating its architecture and design decisions,</li><li>analysis of the current Move language restrictions, ABI and understanding if forking the language is still necessary,</li><li>analysis of the potential effects of forking the language and the toolchain if it is deemed necessary.</li></ul>Medium: A markdown design decision document in the repository. |
+| 1. | System Pallet: Substrate Move | We will create a Substrate system pallet that will provide the RPC calls as the initial interfaces for interacting with the Move VM. The Move VM port itself will not be implemented, all of the methods will be empty stubs.</br> Solid code practices will be in place: CI/CD, tests, documentation, linting, and publication of the library to http://crates.io. |
+| 1. | Rust crate: Substrate Move | Forking the Move VM if deemed necessary. The alterations would include everything to create the virtual machine Substrate-compatible. We would also provide extensive documentation of how the whole process is designed and how it is to be maintained. |
```

Comment:
Could you put the documentation and discovery findings for the VM fork into a separate deliverable? I think it's important enough to warrant its own.

Reply:
Thank you, I added the changes you requested. Do they satisfy your expectations?